source/gridfs/tests/download.yml

```diff
@@ -1,192 +1,241 @@
-                -
-                    { update : "fs.chunks", updates : [
-                        { q : { files_id : { "$oid" : "000000000000000000000005" }, n : 2 }, u : { $set : { data : { $hex : "99" } } } }
-                    ] }
-        act:
-            operation: download
-            arguments:
-                id: { "$oid" : "000000000000000000000005" }
-        assert:
-            error: "ChunkIsWrongSize"
-    -
-        description: "Download legacy file with no name"
-        act:
-            operation: download
-            arguments:
-                id: { "$oid" : "000000000000000000000006" }
-                options: { }
-        assert:
-            result: { $hex : "1122" }
+  - description: "download when length is zero"
+    operations:
+      - name: download
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000001" }
+        expectResult: { $$matchesHexBytes: "" }
+  - description: "download when length is zero and there is one empty chunk"
+    operations:
+      - name: download
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000002" }
+        expectResult: { $$matchesHexBytes: "" }
+  - description: "download when there is one chunk"
+    operations:
+      - name: download
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000003" }
+        expectResult: { $$matchesHexBytes: "1122" }
+  - description: "download when there are two chunks"
+    operations:
+      - name: download
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000004" }
+        expectResult: { $$matchesHexBytes: "1122334455667788" }
+  - description: "download when there are three chunks"
+    operations:
+      - name: download
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000005" }
+        expectResult: { $$matchesHexBytes: "112233445566778899aa" }
+  - description: "download when files entry does not exist"
+    operations:
+      - name: download
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000000" }
+        expectError: { isError: true } # FileNotFound
+  - description: "download when an intermediate chunk is missing"
+    operations:
+      - name: deleteOne
+        object: *bucket0_chunks_collection
+        arguments:
+          filter:
+            files_id: { $oid: "000000000000000000000005" }
+            n: 1
+        expectResult:
+          deletedCount: 1
+      - name: download
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000005" }
+        expectError: { isError: true } # ChunkIsMissing
+  - description: "download when final chunk is missing"
+    operations:
+      - name: deleteOne
+        object: *bucket0_chunks_collection
+        arguments:
+          filter:
+            files_id: { $oid: "000000000000000000000005" }
+            n: 2
```

Comment:
The [original test](https://github.com/mongodb/specifications/blob/926241af4090d4345057b966581fd329262bbc3b/source/gridfs/tests/download.yml#L147) used `n: 1`, which was incorrect copypasta from the preceeding [Download when an intermediate chunk is missing](https://github.com/mongodb/specifications/blob/926241af4090d4345057b966581fd329262bbc3b/source/gridfs/tests/download.yml#L128) test.